### DISCUSSION TITLE
Optimize values watch

### DISCUSSION THREAD

**Discussion by lionskape** [react-hook-form/react-hook-form#4420] [Ideas]
Is there any possibility to return the same values object from the watch method, if no values were changed since the previous call?

It would be great to use that object as a dependency in the effects/memoization/etc.
So, now, to achieve the same behavior I have to pass each value of each key as a dependency to the hooks.

**Comment by bluebill1049** [comment]
no, we want `watch` to produce value simple and fast which can be used inside the render. (don't want to memorize value inside the hook)
